system.go

```diff
@@ -52,6 +52,16 @@ func Host() (types.Host, error) {
 	return provider.Host()
 }
 
+// HostFS is the same as Host, but allows for a custom root hostfs mountpoint
+// a custom filesystem root is only supported on linux, and this will fallback to the default provider on other platforms
+func HostFS(hostfs string) (types.Host, error) {
```

Comment:
Have you considered using the option pattern to enhance the existing APIs? My initial thought is that this would better because it keeps the [api surface](https://pkg.go.dev/github.com/elastic/go-sysinfo#section-documentation) of the package small and direct.

Reply:
I actually did think of that, but the idea of making some `...Option` arg for something where there's only ever going to be a single option felt kind of deceptive or non-obvious?